### DISCUSSION TITLE
60x32 0.5" OLED - Help Required

### DISCUSSION THREAD

**Discussion by MightyBenster** [olikraus/u8g2#2369] [Q&A]
Hello All,

I have a few of [these](https://www.dfrobot.com/product-1353.html) 0.5" displays from DFRobot. This library is one of the few that supports it.
I have been using it well on breadboard, but would now like to integrate into my own PCB.

My first iteration has arrived, but the display is the only thing I am not able to get working.
I am using several BGA components that are all working as intended, and cannot find anything untoward in the assembly (shorts, OC, etc).

Would anybody have any input to help my troubleshooting?
![image](https://github.com/olikraus/u8g2/assets/132838483/851d5eae-2d74-47ab-ba23-722085689404)

I expected to be able to see the I2C address using an iterative search - no luck.
I also hoped to communicate directly using the same code that has been working with the provided supply-board (I2C 0x060h).

Any input will be appreciated.

Ben

**Comment by olikraus** [comment]
Does you step up converter work? If I remember correctly, some 13V are required.
The "official" address of the controller is 0x030 if I remember correctly. U8g2 just uses the version, which is multiplied with 2

**Reply by olikraus** [reply]
Just for completeness: Remember to add pull up resistors to the I2C bus lines

Here is the 10 year old schematic, which I have used once to drive that display manually and which was working...

![schematic_ld7032_oled(2)](https://github.com/olikraus/u8g2/assets/1084992/63690b89-10ec-49e9-b0c7-7ebf0ffd6d86)

**Reply by MightyBenster** [reply]
Thank you for your quick reply, and the Schematic.
The only notable difference appears to be an additional 0.1uF Cap on VCC_R.

Tested the Step-up output is giving ~15V, which is within spec.
I assume your HC4050 is to provide 3.3V logic?

**Reply by olikraus** [reply]
> I assume your HC4050 is to provide 3.3V logic?

Yes

**Reply by olikraus** [reply]
> The only notable difference appears to be an additional 0.1uF Cap on VCC_R.

I had also added a Schottky diode, but maybe this is already included in your step up convert IC

**Reply by olikraus** [reply]
One more topic: I think I have never tested I2C with this device. So I am not sure whether u8g2 I2C mode works with this display. The problem is, that I2C bus protocol differs a lot across the graphics devices and as a consequence, it might be wrong.

Did you test the display with I2C mode and u8g2 on your breadboard? If so, then I think it should be ok.

The implementation of the I2C bus protocol for the LD7032 is here: 

https://github.com/olikraus/u8g2/blob/8430f65c9e2bc7ca393bf0e9d8cf58a84ae081d7/csrc/u8x8_cad.c#L657

**Reply by MightyBenster** [reply]
Olikraus, Thank you for your help!

There were several issues, but mostly stupid (poor PCB connection, driver auto-update, etc).
The RESET line could not be none, and must be DisplayRST pin:

`U8G2_LD7032_60X32_F_SW_I2C u8g2(U8G2_R0, /* clock=*/ 2, /* data=*/ 1, /* reset=*/ U8X8_PIN_NONE);`
`U8G2_LD7032_60X32_F_SW_I2C u8g2(U8G2_R0, /* clock=*/ 2, /* data=*/ 1, /* reset=*/ 39);`


This was not the same on the supplied driver board, so I maybe have missed something in design - but it works, so I will not make confusion!

Thank you again,

Ben
